[Clang] Reuse the function getOptimizationLevel in tools::addLTOOptions.

tclin914 opened the pull request.

tclin914 committed:
[Clang] Reuse the function getOptimizationLevel in tools::addLTOOptions.

I also move the functions getOptimizationLevel/getOptimizationLevelSize
to Driver.cpp. That we can more easily use those functions. Don't need
to add clangFrontend to LINK_LIBS additionally.

qiongsiwu reviewed (commented):
Thanks for working on the PR! It looks good to me overall but I have one question.

tclin914 committed:
Move getOptimizationLevel/getOptimizationLevelSize to `clang::driver` namespace

tclin914 commented on clang/include/clang/Driver/Driver.h:
```diff
@@ -916,6 +922,12 @@ void applyOverrideOptions(SmallVectorImpl<const char *> &Args,
+                              DiagnosticsEngine &Diags);
+
+unsigned getOptimizationLevelSize(const llvm::opt::ArgList &Args);
+
```
Move these functions to `clang::driver` namespace. Thanks.

Thibault-Monnier reviewed (changes requested):
Please fix the compilation.

Thibault-Monnier reviewed (approved):
LGTM after you fix the compilation.

tclin914 committed:
Don't move the functions to Driver

After https://github.com/llvm/llvm-project/pull/169599, the dependency
of clangDriver and clangFrontend is removed.

tclin914 committed:
Revert "Move getOptimizationLevel/getOptimizationLevelSize to `clang::driver` namespace"

This reverts commit 0fc8606c9555cab027b13907737c72006ab04524.

tclin914 committed:
Revert "[Clang] Reuse the function getOptimizationLevel in tools::addLTOOptions."

This reverts commit c15ccbb6f797c24d9a3b28f5e0d8d9ca93a76668.
